### PR TITLE
add conv1d layer type (used in gpt2)

### PR DESCRIPTION
**Pull request by zywilliamli**
gpt2 type models use transformers.pytorch_utils.Conv1D instead of nn.Linear which is not currently tracked by gradient collector, this pr adds that layer type. 

tested manually using python -m bergson runs/test --model openai-community/gpt2 --dataset NeelNanda/pile-10k --truncation, also added unit tests for it.

### CODE REVIEW

File: bergson/gradients.py [hunk 1]
```diff
@@ -442,15 +443,21 @@ def _save_input(self, module: nn.Module, inp: tuple, _):
         # to save memory, rather than waiting until the backward pass.
         p = self.processor.projection_dim
         if p is not None and not isinstance(norm, AdamNormalizer):
-            i = module.in_features
+            # Handle both Linear and Conv1D modules
```
Comment (luciaquirke) [inline bergson/gradients.py]: Unnecessary comment

File: bergson/gradients.py [hunk 2]
```diff
@@ -442,15 +469,15 @@ def _save_input(self, module: nn.Module, inp: tuple, _):
...
+            i = getattr(module, LayerAdapter.in_func(module))
             x = x @ self.projection(name, p, i, "right", x.device, x.dtype).T  # type: ignore
 
         module._inputs = x
 
     def _process_grad(self, module: nn.Module, _, grad_out):
         """Process the incoming gradient wrt the output of the module."""
         # Sanity checks
-        assert isinstance(module, nn.Linear), "Expected a Linear module"
+        assert isinstance(module, LayerAdapter.supported_modules), "Expected a supported module"
```
Comment (luciaquirke) [inline bergson/gradients.py]: Could we print the supported modules here?

**Review by luciaquirke — APPROVED** [review]
LGTM 🚀
